File: output/ChatGPT/avatar/Python/C++/codeforces_70_B.cpp
#include <iostream>
#include <regex>
using namespace std;

int main() {
    int n;
    cin >> n;
    string input;
    getline(cin.ignore(), input);
    
    int ans = 1, sumL = 0;
    regex delimiter("[.?!]");
    sregex_token_iterator it(input.begin(), input.end(), delimiter, -1), end;
    
    for (; it != end; ++it) {
        string s = it->str();
        s = regex_replace(s, regex("^\\s+"), "") + ".";
        int L = s.length();
        
        if (L > 1) {
            if (L > n) {
                cout << "Impossible" << endl;
                return 0;
            }
            if (sumL + L + (sumL > 0) > n) {
                ans += 1;
                sumL = L;
            } else {
                sumL = sumL + L + (sumL > 0);
            }
        }
    }
    cout << ans << endl;

    return 0;
}

File: output/ChatGPT/avatar/Python/Java/codeforces_70_B.java
import java.util.Scanner;
import java.util.regex.Pattern;

public class codeforces_70_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = Integer.parseInt(scanner.nextLine());
        int ans = 1;
        int sumL = 0;

        String input = scanner.nextLine();
        String[] sentences = Pattern.compile("[.?!]").split(input);

        for (String s : sentences) {
            s = s.trim() + ".";
            int L = s.length();

            if (L > 1) {
                if (L > n) {
                    System.out.println("Impossible");
                    System.exit(0);
                }

                if (sumL + L + (sumL > 0) > n) {
                    ans++;
                    sumL = L;
                } else {
                    sumL = sumL + L + (sumL > 0);
                }
            }
        }

        System.out.println(ans);
    }
}